src/sentry/web/frontend/base.py

```diff
@@ -487,8 +488,8 @@ def is_auth_required(
 
         return False
 
-    def handle_permission_required(self, request: Request, organization: Organization | RpcOrganization, *args: Any, **kwargs: Any) -> HttpResponse:  # type: ignore[override]
-        if self.needs_sso(request, organization):
+    def handle_permission_required(self, request: Request, organization: Organization | RpcOrganization | None, *args: Any, **kwargs: Any) -> HttpResponse:  # type: ignore[override]
```

Comment:
I've adjusted the `organization` type here, since `handle_permission_required()` will be called when an organization is not visible.